sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/DeserializerBuildHelper.scala

```diff
@@ -270,6 +270,8 @@ object DeserializerBuildHelper {
       enc: AgnosticEncoder[_],
       path: Expression,
       walkedTypePath: WalkedTypePath): Expression = enc match {
+    case ae: AgnosticExpressionPathEncoder[_] =>
+      ae.fromCatalyst(path)
```

Comment:
Is this tested anywhere? I know you have a test for` toCatalyst`.

Reply:
The "SPARK-49960: joinWith custom encoder" in DatasetSuite tests it
```scala
val ds2 = spark.createDataset[ClassData](Seq(ClassData("a", 1),
      ClassData("b", 2)))(CustomPathEncoder.custClassDataEnc)
```